lib/useDeferredList/specs/useDeferredList.spec.js

```diff
@@ -20,10 +20,23 @@ describe('useDeferredList', () => {
     act(() => result.current.loadItems());
 
     expect(result.current.deferredList).toEqual(times(15, index => index));
+  });
 
-    act(() => result.current.loadItems());
+  it('should load more items if the initialItems argument is updated to a higher number than the current item count', () => {
+    let initialItemCount = 10;
+
+    const { result, rerender } = renderHook(
+      () => useDeferredList({ initialItemCount, list, numberOfItemsToLoad: 5 }),
+      {}
+    );
 
-    expect(result.current.deferredList).toEqual(times(20, index => index));
+    expect(result.current.deferredList).toEqual(times(10, index => index));
+
+    initialItemCount = 15;
+    rerender();
+    rerender();
```

Comment:
Why the double rerender here? I don't know, but the test doesn't work without it.

Once to update the `initialCount` value, but why the second time?

Reply:
Has a timeout or event been introduced maybe? 🤔

Reply:
> Has a timeout or event been introduced maybe? 🤔
No timeouts or events, purely setting state and useEffects!
> This is weird, what happens without it?
With just one `rerender` it doesn't perform the final state update, and the list remains at it's previous value (and the test fails)

I'll add this to a debt card in the release, as it does bug me!